src/ert/ensemble_evaluator/dispatch.py

```diff
@@ -1,107 +1,106 @@
+            target=self.run_dispatcher,
+        )
+        self._dispatcher_thread.start()
 
-    async def _work(self):
+    def _work(self):
         if len(self._buffer) == 0:
             logger.debug("no events to be processed in queue")
             return
 
-        t0 = time.time()
-        batch_of_events_for_processing, self._buffer = (
-            self._buffer[: self._max_batch],
-            self._buffer[self._max_batch :],
-        )
-        left_in_queue = len(self._buffer)
-
+        with self._buffer_lock:
+            t0 = time.time()
+            batch_of_events_for_processing, self._buffer = (
+                self._buffer[: self._max_batch],
+                self._buffer[self._max_batch :],
+            )
+            left_in_queue = len(self._buffer)
         function_to_events_map = OrderedDict()
         for f, event in batch_of_events_for_processing:
             if f not in function_to_events_map:
                 function_to_events_map[f] = []
             function_to_events_map[f].append(event)
 
-        def done_logger(_):
-            logger.debug(
-                f"processed {len(batch_of_events_for_processing)} events in "
-                f"{(time.time()-t0):.6f}s. "
-                f"{left_in_queue} left in queue"
-            )
+        for f, events in function_to_events_map.items():
+            f(events)
 
-        events_handling = asyncio.gather(
-            *[f(events) for f, events in function_to_events_map.items()]
+        logger.debug(
+            f"processed {len(batch_of_events_for_processing)} events in "
+            f"{(time.time()-t0):.6f}s. "
+            f"{left_in_queue} left in queue"
         )
-        events_handling.add_done_callback(done_logger)
-        await events_handling
 
-    async def _job(self):
-        while self._running:
-            await asyncio.sleep(self._timeout)
-            await self._work()
+    def run_dispatcher(self):
+        try:
+            self._job()
+        except Exception as failure:  # pylint: disable=broad-exception-caught
+            logger.exception(f"exception in batching dispatcher: {failure}")
+            trace_info = traceback.format_exception(
+                type(failure), failure, failure.__traceback__
+            )
+            logger.error(f"{trace_info}")
+        else:
+            logger.debug("batcher finished normally")
+            return
+        finally:
+            self._finished = True
+
+        # call any registered handlers for FAILED. since we don't have
+        # an event, pass empty list and let handler decide how to proceed
+        failure_func = self._LOOKUP_MAP[identifiers.EVTYPE_ENSEMBLE_FAILED]
+        failure_func([])
 
+        logger.debug("Dispatcher thread exiting.")
+
+    def _job(self):
+        while self._running:
+            if len(self._buffer) < self._max_batch:
+                time.sleep(self._sleep_between_batches_seconds)
+            else:
+                time.sleep(0)
```

Comment:
`sleep(0)` needs some explanation.

Reply:
sleep(0) is a way to play nice with other threads. you give control back to whatever scheduler is running and let it choose which thread to run next.

Reply:
I guess this comes from the empirical experience to prevent the constant processing execution. I guess you can shed more light into it @valentin-krasontovitsch

Reply:
I added the if else so that we do not spend 2 seconds doing nothing if our queue is already full. This change could possibly have negative consequences as it will also cause more frequent updates in gui, but @valentin-krasontovitsch has another PR that makes the gui updates several orders of magnitude quicker.

The sleep(0) I added because AFAIK it is considered good practice to give the scheduler a possibility to switch to other running threads in between work.

Reply:
Added a comment in source code about the sleep